### PR TITLE
Improve imaplib return types

### PR DESCRIPTION
**Pull request by srittau**
Mark CommandResults as obsolete.

Also fix types of tagged_commands and untagged_responses.

Based on a discussion in #3655.

### CODE REVIEW

File: stdlib/2and3/imaplib.pyi [hunk 1]
```diff
@@ -19,8 +25,8 @@ class IMAP4:
     debug: int = ...
     state: str = ...
     literal: Optional[Text] = ...
-    tagged_commands: Dict[str, str] = ...
-    untagged_responses: Dict[str, str] = ...
+    tagged_commands: Dict[str, Optional[bytes]]
```
Comment (hauntsaninja) [inline stdlib/2and3/imaplib.pyi]: I think the keys for this are bytes as well, https://github.com/python/cpython/blob/master/Lib/imaplib.py#L1195

Comment (srittau) [reply stdlib/2and3/imaplib.pyi]: Good catch, thanks!

File: stdlib/2and3/imaplib.pyi [hunk 2]
```diff
@@ -19,8 +25,8 @@ class IMAP4:
     debug: int = ...
     state: str = ...
     literal: Optional[Text] = ...
-    tagged_commands: Dict[str, str] = ...
-    untagged_responses: Dict[str, str] = ...
+    tagged_commands: Dict[bytes, Optional[bytes]]
```
Comment (JelleZijlstra) [inline stdlib/2and3/imaplib.pyi]: The value type seems to be some tuple, because there is a line `self.tagged_commands[tag] = (typ, [dat])`.

File: stdlib/2and3/imaplib.pyi [hunk 3]
```diff
@@ -8,8 +6,16 @@ from socket import socket as _socket
...
 from typing import Any, Callable, Dict, IO, List, Optional, Pattern, Text, Tuple, Type, Union
 
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+# TODO: Commands should use their actual return types, not this type alias.
+#       E.g. Tuple[Literal["OK"], List[bytes]]
 CommandResults = Tuple[str, List[Any]]
```
Comment (JelleZijlstra) [inline stdlib/2and3/imaplib.pyi]: This should also be private; it doesn't exist at runtime.

File: stdlib/2and3/imaplib.pyi (hunk 2 above)
Comment (srittau) [reply stdlib/2and3/imaplib.pyi]: Good catch.

**Comment by srittau** [comment]
@JelleZijlstra I tried to figure the response types with a mix of experiments and reading the (rather hard to follow) code. I am reasonably content that they are correct.

**Comment by srittau** [comment]
I am a bit confused, why CI is even running stubtest, as this branch precedes stubtest. .travis.yml for this branch does not include stubtest and there is no tests/stubtest_whitelists directory to reference. Anyway, I am restarting CI.
